requirement/src/main/java/org/arquillian/cube/requirement/Requirements.java

```diff
@@ -26,4 +33,24 @@ public static void checkRequirement(Requires requires, Annotation context) throw
             }
         }
     }
+
+    private static Requirement getRequirement(Annotation context) {
+        Requirement requirement = null;
+
+        final ServiceLoader<Requirement> requirementLoader = ServiceLoader.load(Requirement.class);
+
+        final Iterator<Requirement> requirements = requirementLoader.iterator();
+
+        while (requirement == null && requirements.hasNext()) {
+            Requirement req = requirements.next();
+            try {
+                req.getClass().getDeclaredMethod("check", context.annotationType());
+                requirement = req;
+            } catch (NoSuchMethodException e) {
+                // if method not found with required signature. Look for next implementation
```

Comment:
What if after iterating over all implementations found on the CP we still have `requirement` being `null`? Shall we fail in such a case?

Reply:
yes we should fail in such case. Good catch